### PR TITLE
chore(d.ts): type runPartial + finishRun

### PR DESCRIPTION
**Pull request by WilcoFiers**
Add types for:

- axe.utils.getFrameContexts
- axe.runPartial
- axe.finishRun

Related to #3059

### CODE REVIEW

File: axe.d.ts [hunk 1]
```diff
@@ -241,9 +243,35 @@ declare namespace axe {
     helpUrl: string;
     tags: string[];
   }
+  interface SerialDqElement {
+    source: string
+    nodeIndexes: number[]
+    selector: number[],
+    xpath: number[],
+    ancestry: number[],
```
Comment (straker) [inline axe.d.ts:251]: These all should be strings, and `selector` and `ancestry` should account for shadow DOM iframes.

```suggestion
    selector: Selector,
    xpath: string[],
    ancestry: Selector,
```

File: axe.d.ts [hunk 2]
```diff
@@ -241,9 +243,35 @@ declare namespace axe {
...
+  interface PartialResult {
+    frames: SerialDqElement[]
+    results: PartialRuleResult[]
+  }
+  interface FrameContext {
+    frameSelector: Selector,
+    frameContext: ContextObject
+  }
+  interface Utils {
+    getFrameContexts: (context: ElementContext | null) => FrameContext[]
```
Comment (straker) [inline axe.d.ts:269]:
```suggestion
    getFrameContexts: (context?: ElementContext) => FrameContext[]
```

**Review by straker — CHANGES_REQUESTED** [review]

File: axe.d.ts [hunk 3]
```diff
@@ -241,9 +243,35 @@ declare namespace axe {
...
+    selector: number[],
+    xpath: number[],
+    ancestry: number[],
+  }
+  interface PartialRuleResult {
+    id: string,
+    result: 'inapplicable',
+    pageLevel: boolean,
+    impact: null,
+    nodes: object[],
```
Comment (stephenmathieson) [inline axe.d.ts:258]: The `object` type is extremely difficult to use/not what we want.

```suggestion
    nodes: Array<Record<string, unknown>>,
```

**Review by stephenmathieson — CHANGES_REQUESTED** [review]

**Review by stephenmathieson — DISMISSED** [review]
LGTM. Should wait for @straker for the final OK though

**Review by straker — APPROVED** [review]

**Review by stephenmathieson — APPROVED** [review]

**Review by straker — APPROVED** [review]

**Pull request merged**